File: Business/TravelRepublic.Business/Engines/Requests/HotelSearchFilterRequest.cs
using Eml.Mediator.Contracts;
using TravelRepublic.Business.Engines.Responses;

namespace TravelRepublic.Business.Engines.Requests
{
    public class HotelSearchFilterRequest : HotelSearchRequestBase, IRequestAsync<HotelSearchFilterRequest, HotelSearchFilterResponse>
    {
        public HotelSearchFilterRequest(string name, int star, double userRating, double costMin, double costMax)
            : base(name, star, userRating, costMin, costMax)
        {
        }
    }
}

File: Hosts/TravelRepublic.Spa/src/app/modules/hotels/requests/hotel-search-filter-request.ts
import { HotelSearchRequestBase } from './hotel-search-request-base';

export class HotelSearchFilterRequest extends HotelSearchRequestBase {
    constructor(public name = '',
                public star = 0,
                public userRating = 0,
                public costMin = 0,
                public costMax = 0) {
        super(name, star, userRating, costMin, costMax);
    }
}
